src/Symfony/Component/OptionsResolver/Tests/OptionsResolverTest.php

```diff
@@ -569,6 +570,9 @@ public function provideInvalidTypes()
             [42, 'string', 'The option "option" with value 42 is expected to be of type "string", but is of type "integer".'],
             [null, 'string', 'The option "option" with value null is expected to be of type "string", but is of type "NULL".'],
             ['bar', '\stdClass', 'The option "option" with value "bar" is expected to be of type "\stdClass", but is of type "string".'],
+            [['foo', 12], 'string[]', 'The option "option" with value array is expected to be of type "string[]", but one of the elements is of type "integer".'],
+            [123, ['string[]', 'string'], 'The option "option" with value 123 is expected to be of type "string[]" or "string", but is of type "integer".'],
+            [[123], ['string[]', 'string'], 'The option "option" with value array is expected to be of type "string[]" or "string", but is of type "integer|array".'],
```

Comment:
when the value is an array and the allowed type (or the last verified type) is scalar the message is not accurate again, instead I'd expect this:
```
The option "option" with value array is expected to be of type "string[]" or "string", but is of type "integer[]|array".
```
even better if we could remove the `|array` part or show only `"array"` would also be fine.